pkg/types/var.go

```diff
@@ -31,18 +32,46 @@ type Var struct {
 	// purview of this kustomization. ObjRef should use the
 	// raw name of the object (the name specified in its YAML,
 	// before addition of a namePrefix).
-	ObjRef corev1.ObjectReference `json:"objref" yaml:"objref"`
+	ObjRef Target `json:"objref" yaml:"objref"`
 
 	// FieldRef refers to the field of the object referred to by
 	// ObjRef whose value will be extracted for use in
 	// replacing $(FOO).
-	// If unspecified, this defaults to fieldpath: metadata.name
-	FieldRef corev1.ObjectFieldSelector `json:"fieldref,omitempty" yaml:"objref,omitempty"`
+	// If unspecified, this defaults to fieldPath: metadata.name
+	FieldRef FieldSelector `json:"fieldref,omitempty" yaml:"fieldref,omitempty"`
+}
+
+// Target refers to a kubernetes object by Group, Version, Kind and Name
+type Target struct {
+	APIVersion string `json:"apiVersion,omitempty" yaml:"apiVersion,omitempty"`
+	gvk.Gvk
+	Name string `json:"name" yaml:"name"`
+}
+
+// FieldSelector contains the fieldPath to an object field, such as metadata.name
```

Comment:
A struct with one field looks odd (could do `type FieldSelector string` instead).
Needs a comment explaining it.

Reply:
This is to keep the backward compatibility of `vars`.